plugins/reindex/src/main/java/org/elasticsearch/plugin/reindex/BulkIndexByScrollResponse.java

```diff
@@ -0,0 +1,146 @@
+package org.elasticsearch.plugin.reindex;
+
+import static java.lang.Math.min;
+import static java.util.Collections.unmodifiableList;
+import static org.elasticsearch.plugin.reindex.BulkIndexByScrollResponse.Fields.BATCHES;
+import static org.elasticsearch.plugin.reindex.BulkIndexByScrollResponse.Fields.FAILURES;
+import static org.elasticsearch.plugin.reindex.BulkIndexByScrollResponse.Fields.NOOPS;
+import static org.elasticsearch.plugin.reindex.BulkIndexByScrollResponse.Fields.TOOK;
+import static org.elasticsearch.plugin.reindex.BulkIndexByScrollResponse.Fields.UPDATED;
+import static org.elasticsearch.plugin.reindex.BulkIndexByScrollResponse.Fields.VERSION_CONFLICTS;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.bulk.BulkItemResponse.Failure;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentBuilderString;
+
+/**
+ * Response used for actions that index many documents using a scroll request.
+ */
+public class BulkIndexByScrollResponse extends ActionResponse implements ToXContent {
+    private long took;
+    private long updated;
+    private int batches;
+    private long versionConflicts;
+    private long noops;
+    private List<Failure> failures;
+
+    public BulkIndexByScrollResponse() {
+    }
+
+    public BulkIndexByScrollResponse(long took, long updated, int batches, long versionConflicts, long noops, List<Failure> failures) {
+        this.took = took;
+        this.updated = updated;
+        this.batches = batches;
+        this.versionConflicts = versionConflicts;
+        this.noops = noops;
+        this.failures = failures;
```

Comment:
When deserializing we wrap the list with `Collections.unmodifiableList()`. Do you think it makes sense to this this here too?